lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -565,7 +565,7 @@ void LegalizeMD::LegalizeLDIMM(IR::Instr * instr, IntConstType immed)
         // Short-circuit 64-bit logical constants
         if (EncoderMD::CanEncodeLogicalConst(immed, 8))
         {
-            Assert(instr->GetDst()->GetSize() == 8 || immed == (uint32)immed);
+            Assert(instr->GetDst()->GetSize() == 8 || immed == (int32)immed);
             instr->GetDst()->SetType(TyInt64);
```

Comment:
This assert is still wrong. In fact the original one was wrong (my bad). We should only do the CanEncodeLogicalConst() check if GetSize() == 8. Then the assert becomes irrelevant.